### ISSUE TITLE
Installing C# Dependencies when debugging node project.

### ISSUE THREAD

**Issue by MicahZoltu** [OmniSharp/omnisharp-vscode#2536]
## Environment data
`dotnet --info` output:
```
.NET Core SDK (reflecting any global.json):
 Version:   2.1.302
 Commit:    9048955601

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.17134
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\2.1.302\

Host (useful for support):
  Version: 2.1.2
  Commit:  811c3ce6c0

.NET Core SDKs installed:
  2.1.200 [C:\Program Files\dotnet\sdk]
  2.1.302 [C:\Program Files\dotnet\sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.All 2.1.2 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.2 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.0.7 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.2 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]

To install additional .NET Core runtimes or SDKs:
  https://aka.ms/dotnet-download
```
VS Code version:
```
[Window Title]
Visual Studio Code

[Main Instruction]
Visual Studio Code

[Content]

Version: 1.27.2 (user setup)
Commit: f46c4c469d6e6d8c46f268d1553c5dc4b475840f
Date: 2018-09-12T16:17:45.060Z
Electron: 2.0.7
Chrome: 61.0.3163.100
Node.js: 8.9.3
V8: 6.1.534.41
Architecture: x64

[OK] [Copy]
```
C# Extension version: `1.16.1`

## Steps to reproduce
1. Open a node project.
2. Configure a `type: 'node'` project in `launch.json`.
3. Press F5 to begin debugging.
4. --> Notice that Output window says it is installing C# dependencies.
5. Disable C# extension.
6. Repeat step (3).
7. --> Notice that the project correctly starts debugging.

## Expected  behavior
When debugging a non-C# project, C# extension doesn't install a bunch of stuff onto my system before letting me debug my project.

## Actual behavior
C# extension interrupts my process and makes me install a bunch of stuff that I will not be using in current debugging session.

**Comment by rchande** [comment]
@MicahZoltu Do you only see this when you start debugging? Eg, not directly after opening the directory in VS Code? (cc @gregg-miskelly )

**Comment by MicahZoltu** [comment]
Yes.  I can open VSCodo and open projects without seeing this.  When I try to debug a node project, I see it instead of my node project debugging.

**Comment by gregg-miskelly** [comment]
@MicahZoltu the issue is that you haven't successfully downloaded the C# extension dependencies before (it is a 1-time step that happens on the first activation after the C# extension is installed). Ideally this download step would happen as part of the primary extension download, but unfortunately VS code doesn't support that. The way VS Code activation works we need to activate our extension on start debugging, and from what I recall, we can't condition it on what type of thing you are debugging.

The good news is that this should be very rare (after the download succeeds you shouldn't see it again until a new version of the C# extension comes out).

Let us know if you are seeing it frequently and we can try to help troubleshoot. Otherwise, there isn't anything we can do to improve this.

**Comment by RevealedFrom** [comment]
I am facing the same phenomenon.
I created a Node.js project from scratch. When I pressed F5, before configuring any run task, it went into:

`Installing C# dependencies...
Platform: win32, x86_64

Downloading package 'OmniSharp for Windows (.NET 4.6 / x64)' (31021 KB).................... Done!
Installing package 'OmniSharp for Windows (.NET 4.6 / x64)'

Downloading package '.NET Core Debugger (Windows / x64)' (43046 KB).................... Done!
Installing package '.NET Core Debugger (Windows / x64)'

Downloading package 'Razor Language Server (Windows / x64)' (46894 KB).................... Done!
Installing package 'Razor Language Server (Windows / x64)'

Finished`
